monai/losses/dice.py

```diff
@@ -147,7 +147,9 @@ def forward(self, input: torch.Tensor, target: torch.Tensor, smooth: float = 1e-
 
 class MaskedDiceLoss(DiceLoss):
     """
-    Same as DiceLoss, but accepts a binary mask ([0,1]) indicating a region over which to compute the dice.
+    Add an additional `masking` progress before `DiceLoss`, accept a binary mask ([0, 1]) indicating a region,
+    `input` and `target` will be masked by the region: region with mask `1` will keep the original value,
+    region with `0` mask will be converted to `0`. Then feed `input` and `target` to normal `DiceLoss` computation.
```

Comment:
I would append something like "This has the effect of ensuring only the masked region contributes to the loss computation and hence gradient calculation." Do we have some motivating use case where this is useful? It would help to explain why one would use a particular loss function over another.

Reply:
Actually, I think this function was added as kind of "attention" mechanism, it still computes the unmasked region as `0` values, not only the masked region region contributes to the loss computation.
What do you think?
Thanks.

Reply:
If unmasked regions are 0 I wouldn't think they really contributing to the gradient. Is the mask supposed to be learnable? I don't see what the use case would be of this still.

Reply:
I double-checked the source code, I think you are right. Updated according to your comments.
And I think maybe you can check the original bug: https://github.com/Project-MONAI/MONAI/issues/657?, @wyli may know more about the usage example.

Thanks.

Reply:
I think @wyli should check this then, the usage described in the bug report is along the lines of what I was thinking in which case we haven't fixed the bug. If this usage isn't the intended one then we need to clarification on what the attention mechanism is.

Reply:
Sure, @wyli , could you please help share some comments?
Thanks.

Reply:
yes my understanding is that the mask tensor could be learnable, a similar approach ("voxel-level difficulty-aware attention") is used in this recent IJCV paper: https://link.springer.com/article/10.1007/s11263-020-01321-2

Reply:
@wyli , so you suggest changing the existing `MaskedDiceLoss` instead of just updating its doc-string?
If yes, maybe we should discuss this with @myron ?
Thanks.

Reply:
i think updating the docstring would be good enough, but perhaps @myron could review the issue and this PR

Reply:
OK, sure, let me invite @myron for a review here.
Thanks.